### ISSUE TITLE
C# Clientgenerator dont work when server send HttpResponseMessage with the object inside

### ISSUE THREAD

**Issue by edubenito** [swagger-api/swagger-codegen#2684]
this web api method
```
[Authorize(Roles = Constants.ADMIN_ROL + "," + Constants.SEC_ADMIN_ROL)]
        public HttpResponseMessage Put([FromBody]ApplicationVM application)
        {
            try
            {

                var mapper = _applicationMapperConfigurationVm.CreateMapper();
                var applicationModel = _applicationService.Find(application.APP_Id);
                
                if (applicationModel == null)
                    return Request.CreateResponse(HttpStatusCode.NotFound, application);

                applicationModel = mapper.Map(application, applicationModel);
                _applicationService.Update(applicationModel);

                var records = _unitOfWork.SaveChanges(RequestUser, RequestedFrom);
                if (records <= 0) return Request.CreateResponse(HttpStatusCode.NotModified);

                var mapperVm = _applicationMapperConfigurationMv.CreateMapper();
                application = mapperVm.Map<Application, ApplicationVM>(applicationModel);
                return Request.CreateResponse(HttpStatusCode.OK, application);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.BadRequest, ex);
            }
        }
```

produce this generated code
```
public Object PutByApplication (ApplicationVM application)
        {
            ..
            ...
**return (Object) ApiClient.Deserialize(response.Content, typeof(Object), response.Headers);**
        }
```

the result object is a "string" with the json instead of return an ApplicationVM, is there a way yo can improve this.

maybe you can add in the Model a constructor with the Json. or add a static method oposite to this one
 public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

**Comment by wing328** [comment]
@edubenito can you share the spec with us to reproduce the issue?

**Comment by edubenito** [comment]
{"swagger":"2.0","info":{"version":"v1","title":"XXXXXX Conductor API V1","description":"XXXXXX Conductor API"},"host":"webwp01v01d.dcs.XXXXXXclub.es","basePath":"/XXXXXXConductorWS","schemes":["http"],"paths":{"/api/v1/sec/Applications":{"get":{"tags":["ApplicationsController"],"summary":"Get All Applicatios that match the filters","operationId":"GetList","consumes":[],"produces":["application/json","text/json"],"parameters":[{"name":"appName","in":"query","description":"if null filter does not apply, otherwhise perform contains search","required":false,"type":"string"},{"name":"appActive","in":"query","description":"if null filter does not apply, otherwhise perform equallity search Accepted Values (true, false)","required":false,"type":"string"},{"name":"orderBy","in":"query","description":"if null order by APP_name Accepted Values list \",\" sepated with fields order example:\"+APP_Name,-APP_Description\"","required":false,"type":"string"},{"name":"page","in":"query","description":"Offset records, if null all records returned, otherwhise check \"X-Total-Count\" header to retrieve the total records","required":false,"type":"integer","format":"int32"},{"name":"pageSize","in":"query","description":"Size of the Page required if page is set to a value other than null","required":false,"type":"integer","format":"int32"}],"responses":{"200":{"description":"OK","schema":{"type":"array","items":{"$ref":"#/definitions/ApplicationVM"}}}},"deprecated":false},"put":{"tags":["ApplicationsController"],"summary":"Update the Application table","description":"Update the Application table","operationId":"PutByApplication","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"application","in":"body","description":"New Application values","required":true,"schema":{"$ref":"#/definitions/ApplicationVM"}}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/Object"}},"304":{"description":"Not Modified"},"400":{"description":"Bad request"},"404":{"description":"Not Found"}},"deprecated":false},"post":{"tags":["ApplicationsController"],"summary":"Create  new Application","operationId":"PostByApplication","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"application","in":"body","description":"Application to be created","required":true,"schema":{"$ref":"#/definitions/ApplicationVM"}}],"responses":{"201":{"description":"Created","schema":{"$ref":"#/definitions/Object"}},"400":{"description":"Bad request"}},"deprecated":false},"delete":{"tags":["ApplicationsController"],"summary":"Delete Application","operationId":"DeleteByApplication","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"application","in":"body","description":"Application to be deleted","required":true,"schema":{"$ref":"#/definitions/ApplicationVM"}}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/Object"}},"400":{"description":"Bad request"},"404":{"description":"Not Found"}},"deprecated":false}},"/api/v1/sec/Applications/{id}":{"get":{"tags":["ApplicationsController"],"summary":"Get the Application by its primary key","operationId":"GetById","consumes":[],"produces":["application/json","text/json"],"parameters":[{"name":"id","in":"path","description":"APP_Id","required":true,"type":"integer","format":"int32"}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/ApplicationVM"}}},"deprecated":false}},"/api/v1/sec/AppModuleOptions":{"get":{"tags":["AppModuleOptionsController"],"summary":"Get All AppModuleOptions that match the filters","operationId":"GetList","consumes":[],"produces":["application/json","text/json"],"parameters":[{"name":"amId","in":"query","description":"Id of the AppModuleOption Parent","required":false,"type":"integer","format":"int32"},{"name":"amoName","in":"query","description":"if null filter does not apply, otherwhise perform contains search","required":false,"type":"string"},{"name":"amoActive","in":"query","description":"if null filter does not apply, otherwhise perform equallity search Accepted Values (true, false)","required":false,"type":"string"},{"name":"orderBy","in":"query","description":"if null order by AMO_Name Accepted Values list \",\" sepated with fields order example:\"+AM_Name,-AM_Description\"","required":false,"type":"string"},{"name":"page","in":"query","description":"Offset records, if null all records returned, otherwhise check \"X-Total-Count\" header to retrieve the total records","required":false,"type":"integer","format":"int32"},{"name":"pageSize","in":"query","description":"Size of the Page required if page is set to a value other than null","required":false,"type":"integer","format":"int32"}],"responses":{"200":{"description":"OK","schema":{"type":"array","items":{"$ref":"#/definitions/AppModuleOptionVM"}}}},"deprecated":false},"put":{"tags":["AppModuleOptionsController"],"summary":"Update the AppModuleOption table","description":"Update the AppModuleOption table","operationId":"PutByAppmoduleoption","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"appModuleOption","in":"body","description":"New AppModuleOption values","required":true,"schema":{"$ref":"#/definitions/AppModuleOptionVM"}}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/Object"}},"304":{"description":"Not Modified"},"400":{"description":"Bad request"},"404":{"description":"Not Found"}},"deprecated":false},"post":{"tags":["AppModuleOptionsController"],"summary":"Create  new AppModuleOption","operationId":"PostByAppmoduleoption","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"appModuleOption","in":"body","description":"AppModuleOption to be created","required":true,"schema":{"$ref":"#/definitions/AppModuleOptionVM"}}],"responses":{"201":{"description":"Created","schema":{"$ref":"#/definitions/Object"}},"400":{"description":"Bad request"}},"deprecated":false},"delete":{"tags":["AppModuleOptionsController"],"summary":"Delete AppModuleOption","operationId":"DeleteByAppmoduleoption","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"appModuleOption","in":"body","description":"AppModuleOption to be deleted","required":true,"schema":{"$ref":"#/definitions/AppModuleOptionVM"}}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/Object"}},"400":{"description":"Bad request"},"404":{"description":"Not Found"}},"deprecated":false}},"/api/v1/sec/AppModuleOptions/{id}":{"get":{"tags":["AppModuleOptionsController"],"summary":"Get the AppModuleOption by its primary key","operationId":"GetById","consumes":[],"produces":["application/json","text/json"],"parameters":[{"name":"id","in":"path","description":"APP_Id","required":true,"type":"integer","format":"int32"}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/AppModuleOptionVM"}}},"deprecated":false}},"/api/v1/sec/AppModules":{"get":{"tags":["AppModulesController"],"summary":"Get All AppModules that match the filters","operationId":"GetList","consumes":[],"produces":["application/json","text/json"],"parameters":[{"name":"appId","in":"query","description":"Id of the AppModule Parent","required":false,"type":"integer","format":"int32"},{"name":"amName","in":"query","description":"if null filter does not apply, otherwhise perform contains search","required":false,"type":"string"},{"name":"amActive","in":"query","description":"if null filter does not apply, otherwhise perform equallity search Accepted Values (true, false)","required":false,"type":"string"},{"name":"orderBy","in":"query","description":"if null order by AM_Name Accepted Values list \",\" sepated with fields order example:\"+AM_Name,-AM_Description\"","required":false,"type":"string"},{"name":"page","in":"query","description":"Offset records, if null all records returned, otherwhise check \"X-Total-Count\" header to retrieve the total records","required":false,"type":"integer","format":"int32"},{"name":"pageSize","in":"query","description":"Size of the Page required if page is set to a value other than null","required":false,"type":"integer","format":"int32"}],"responses":{"200":{"description":"OK","schema":{"type":"array","items":{"$ref":"#/definitions/AppModuleVM"}}}},"deprecated":false},"put":{"tags":["AppModulesController"],"summary":"Update the AppModule table","description":"Update the AppModule table","operationId":"PutByAppmodule","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"appModule","in":"body","description":"New AppModule values","required":true,"schema":{"$ref":"#/definitions/AppModuleVM"}}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/Object"}},"304":{"description":"Not Modified"},"400":{"description":"Bad request"},"404":{"description":"Not Found"}},"deprecated":false},"post":{"tags":["AppModulesController"],"summary":"Create  new AppModule","operationId":"PostByAppmodule","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"appModule","in":"body","description":"AppModule to be created","required":true,"schema":{"$ref":"#/definitions/AppModuleVM"}}],"responses":{"201":{"description":"Created","schema":{"$ref":"#/definitions/Object"}},"400":{"description":"Bad request"}},"deprecated":false},"delete":{"tags":["AppModulesController"],"summary":"Delete AppModule","operationId":"DeleteByAppmodule","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":["application/json","text/json"],"parameters":[{"name":"appModule","in":"body","description":"AppModule to be deleted","required":true,"schema":{"$ref":"#/definitions/AppModuleVM"}}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/Object"}},"400":{"description":"Bad request"},"404":{"description":"Not Found"}},"deprecated":false}},"/api/v1/sec/AppModules/{id}":{"get":{"tags":["AppModulesController"],"summary":"Get the AppModule by its primary key","operationId":"GetById","consumes":[],"produces":["application/json","text/json"],"parameters":[{"name":"id","in":"path","description":"APP_Id","required":true,"type":"integer","format":"int32"}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/AppModuleVM"}}},"deprecated":false}},"/api/controllers/v1/Countries":{"get":{"tags":["CountriesController"],"operationId":"Get","consumes":[],"produces":["application/json","text/json"],"parameters":[],"responses":{"200":{"description":"OK","schema":{"type":"array","items":{"$ref":"#/definitions/Country"}}}},"deprecated":false},"put":{"tags":["CountriesController"],"operationId":"PutByCountry","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":[],"parameters":[{"name":"country","in":"body","required":true,"schema":{"$ref":"#/definitions/Country"}}],"responses":{"204":{"description":"No Content"}},"deprecated":false},"post":{"tags":["CountriesController"],"operationId":"PostByCountry","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":[],"parameters":[{"name":"country","in":"body","required":true,"schema":{"$ref":"#/definitions/Country"}}],"responses":{"204":{"description":"No Content"}},"deprecated":false},"delete":{"tags":["CountriesController"],"operationId":"DeleteByCountry","consumes":["application/json","text/json","application/x-www-form-urlencoded"],"produces":[],"parameters":[{"name":"country","in":"body","required":true,"schema":{"$ref":"#/definitions/Country"}}],"responses":{"204":{"description":"No Content"}},"deprecated":false}},"/api/controllers/v1/Countries/{id}":{"get":{"tags":["CountriesController"],"operationId":"GetById","consumes":[],"produces":["application/json","text/json"],"parameters":[{"name":"id","in":"path","required":true,"type":"integer","format":"int32"}],"responses":{"200":{"description":"OK","schema":{"$ref":"#/definitions/Country"}}},"deprecated":false}}},"definitions":{"ApplicationVM":{"description":"","type":"object","properties":{"apP_Id":{"format":"int32","description":"","type":"integer"},"apP_Name":{"description":"","type":"string"},"apP_Description":{"description":"","type":"string"},"apP_Active":{"description":"","type":"boolean"}}},"Object":{"type":"object","properties":{}},"AppModuleOptionVM":{"description":"","type":"object","properties":{"amO_Id":{"format":"int32","description":"","type":"integer"},"amO_Name":{"description":"","type":"string"},"amO_Description":{"description":"","type":"string"},"amO_Order":{"format":"int32","description":"","type":"integer"},"amO_View":{"description":"","type":"string"},"amO_Active":{"description":"","type":"boolean"},"amO_AM_Id":{"format":"int32","description":"","type":"integer"}}},"AppModuleVM":{"description":"","type":"object","properties":{"aM_Id":{"format":"int32","description":"","type":"integer"},"aM_Name":{"description":"","type":"string"},"aM_Description":{"description":"","type":"string"},"aM_Order":{"format":"int32","description":"","type":"integer"},"aM_Active":{"description":"","type":"boolean"},"aM_APP_Id":{"format":"int32","description":"","type":"integer"}}},"Country":{"type":"object","properties":{"c_CodeISO3":{"type":"string"},"c_Name":{"type":"string"},"c_CodeISO":{"type":"string"},"c_AlphabeticCodeISO":{"type":"string"},"c_IsEuropeanUnion":{"type":"boolean"},"c_CodeDXS":{"type":"string"},"objectState":{"format":"int32","enum":[0,1,2,3],"type":"integer"}}}}}

Enviado desde mi iPhone

**Comment by edubenito** [comment]
Sorry it was my Fault, I Miss add the Attribute [ResponseType(typeof(XXXX))]

[ResponseType(typeof(ApplicationVM))]
or 
[ResponseType(typeof(IEnumerable<ApplicationVM>))]

Now it worksfine

Apologizes

**Comment by wing328** [comment]
@edubenito no worry. Glad to hear that it's resolved. Btw, we just released stable version 2.1.6 a few weeks ago so please check it out if you've not done so.
